internal/taskexec/execution.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2025 The A2A Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package taskexec
+
+import (
+	"context"
+	"fmt"
+	"iter"
+
+	"github.com/a2aproject/a2a-go/a2a"
+	"github.com/a2aproject/a2a-go/a2asrv/eventqueue"
+)
+
+// Execution represents an agent invocation in a context of the referenced task.
+// If the invocation was finished Result() will resolve immediately, otherwise it will block.
+type Execution struct {
+	tid        a2a.TaskID
+	controller Executor
+
+	subscribeChan   chan chan a2a.Event
+	unsubscribeChan chan chan a2a.Event
+
+	result *promise
+}
+
+// Not exported, because Executions are created by Executor.
+func newExecution(tid a2a.TaskID, controller Executor) *Execution {
+	return &Execution{
+		tid:        tid,
+		controller: controller,
+
+		subscribeChan:   make(chan chan a2a.Event),
+		unsubscribeChan: make(chan chan a2a.Event),
+
+		result: newPromise(),
+	}
+}
+
+// GetEvents subscribes to the events the agent is producing during an active Execution.
+// If the Execution was finished the sequence will be empty.
+func (e *Execution) GetEvents(ctx context.Context) iter.Seq2[a2a.Event, error] {
```

Comment:
Is it possible to name this `Events` and avoid `Get` prefix?

https://pkg.go.dev/iter#hdr-Naming_Conventions